packages/environment-ember-template-imports/-private/environment/preprocess.ts

```diff
@@ -1,26 +1,86 @@
 import { GlintExtensionPreprocess } from '@glint/core/config-types';
-import { parseTemplates } from 'ember-template-imports';
 import { GLOBAL_TAG, PreprocessData, TemplateLocation } from './common';
-
+/**
+ * We don't need to actually generate the code that would be emitted for
+ * a real app to build, we can use placeholders, so we have less offsets
+ * to worry about (precompile, setComponentTemplate, their imports, etc)
+ */
 const TEMPLATE_START = `[${GLOBAL_TAG}\``;
 const TEMPLATE_END = '`]';
 
+// NOTE: This import is a lie -- We are not real ESM, and this is compiled to CJS.
+//       ESM usage would be different here, and this is the CJS version of content-tag.
+import { Preprocessor } from 'content-tag';
+const p = new Preprocessor();
+
+interface Parsed {
```

Comment:
Can we have `content-tag` expose this type itself, either by generating it from the underlying Rust struct(s) or something like [`#[wasm_bindgen(typescript_custom_section)]`](https://rustwasm.github.io/wasm-bindgen/reference/attributes/on-rust-exports/typescript_custom_section.html)?

Reply:
I think so -- tho I don't know how specific we'd get. @ef4 said an update to one of the build tools would give us better types when updated.

It's also totally reasonable for us to ship our own d.ts, and ignore the generated one.

Reply:
I'm not too hung up on any specific approach on the `content-tag` side; it just makes me nervous to be encoding assumptions here that could break at any time 😅